src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -167,6 +167,9 @@ public void onResultNotifiedOld(ProgrammingExerciseParticipation participation)
         }
 
         if (result != null) {
+            // TODO: There is a design issue here: As the participation was not loaded in this session (= Transaction), getExercise will fail when the participation was not loaded
```

Comment:
Can you add a comment with the `@param participation` JavaDoc (Line 149)  to expose this limitation more prominently?

Reply:
You are right, I added some more info.